src/cmds/hardware/i2c_tools/i2cdetect/i2cdetect.c

```diff
@@ -0,0 +1,86 @@
+/**
+ * @file
+ *
+ * @date Nov 16, 2018
+ * @author Anton Bondarev
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <stddef.h>
+
+#include <drivers/i2c/i2c.h>
+
+static void print_usage(void) {
+	printf("Usage: i2cdetect -h\n");
+	printf("Usage: i2cdetect -l\n");
+	printf("Usage: i2cdetect [-y] i2cbus\n");
+	printf("Usage: i2cdetect -F i2cbus\n");
+}
+
+static void print_error(void) {
+	printf("Wrong parameters\n");
+}
+
+static void i2c_bus_list(void) {
+	int i;
+	struct i2c_bus *bus;
+	for (i = 0; i < I2C_BUS_MAX; i++) {
+		bus = i2c_bus_get(i);
+		if (bus) {
+			printf("%s\n", bus->name);
+		}
+	}
+}
+
+static void i2c_bus_scan(long busn) {
+	int i;
+	uint8_t tmp;
+
+	printf("     0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f\n");
+	printf("00:         ");
+	for (i = 0x03; i < 0x77; i++) {
+		if (0 == (i % 0x10)) {
+			printf("\n%2X:", (unsigned)i);
+		}
+		if (0 == i2c_bus_write(busn, i, &tmp, 0)) {
```

Comment:
Is it ok that last argument is zero?

Reply:
Yes, for detect (scan bus) start zero length write operation